download-windows-esd

```diff
@@ -73,54 +73,68 @@ EOT
 download_products_xml
 
 case "$1" in
-    download)
-        if [ -z "$4" ]; then usage; exit 1; fi
-
-        file_xml="$(file_xml "$2" "$3" "$4")"
-
-        filename="$(filename "${file_xml}")"
-        url="$(url "${file_xml}")"
-        expected_sha1="$(sha1 "${file_xml}")"
-
-        curl -#OR -C - "${url}"
-
-        actual_sha1="$(shasum "${filename}" | cut -b -40)"
-        if [ "${expected_sha1}" != "${actual_sha1}" ]; then echo SHA-1 mismatch \(expected ${expected_sha1}\) >&2; exit 5; fi
-        ;;
-
-    shasum)
-        if [ -z "$4" ]; then usage; exit 1; fi
-
-        file_xml="$(file_xml "$2" "$3" "$4")"
-        filename="$(filename "${file_xml}")"
-        expected_sha1="$(sha1 "${file_xml}")"
-
-        echo "${expected_sha1}  ${filename}"
-        ;;
- 
-    url)
-        if [ -z "$4" ]; then usage; exit 1; fi
-
-        file_xml="$(file_xml "$2" "$3" "$4")"
-
-        url="$(url "${file_xml}")"
-        echo "${url}"
-        ;;
-
-    languages)
-        languages
-        ;;
-
-    editions)
-        if [ -z "$2" ]; then usage; exit 1; fi
-        editions "$2"
-        ;;
-
-    architectures)
-        if [ -z "$3" ]; then usage; exit 1; fi
-        architectures "$2" "$3" ;;
-
-    *)
-        usage; exit 1
-        ;; 
+download)
+  if [ -z "$4" ]; then
+    usage
+    exit 1
+  fi
+
+  file_xml="$(file_xml "$2" "$3" "$4")"
+
+  filename="$(filename "${file_xml}")"
+  url="$(url "${file_xml}")"
+  expected_sha1="$(sha1 "${file_xml}")"
+
+  curl -#fSLOR -C - "${url}"
```

Comment:
Do me a favor and explain what each new flag is doing and why we need it? curl's flag list is interminably long and I certainly do not have most of them internalized.

Reply:
Nice. Looks like we do want "-f", do _not_ want "-S" (it's a no-op without "-s", which conflicts with "-O"), and "-L" has not been necessary (Microsoft's CDN has not used redirects) but wouldn't hurt.

So let's remove "-S" then; otherwise this is good 👍